app/models/user.rb

```diff
@@ -13,4 +13,6 @@ class User < ApplicationRecord
   has_many :events
   has_many :attendances, foreign_key: 'attendee_id'
   has_many :events_as_attendee, through: :attendances, source: 'attended_event'
+  scope :past, -> { where("date < ?", DateTime.now) }
```

Comment:
We don't need these scopes in `user` model, past and future events belong to the `event`  model and I can see that you have already wrote it in `event` model.